modules/co2-foam/examples/co2FoamFieldExample.m

```diff
@@ -0,0 +1,172 @@
+dataPath = getDatasetPath('co2_foam');
+fn = fullfile(dataPath, 'field', 'RUN', 'QBOX2_GAS.DATA');
+deck = readEclipseDeck(fn);
+deck = convertDeckUnits(deck);
+
+%% Set up grid, rock, and fluid
+G     = initEclipseGrid(deck);
+G     = computeGeometry(G);
+rock  = initEclipseRock(deck);
+fluid = initDeckADIFluid(deck);
+
+gravity reset on
+
+%% Set up CO2 foam model
+surfactantSystem = 15; % Brij L23
+case_name = 'BOX_K1_C1';
+
+% Mobility reduction function parameters
+foam = getFoam(surfactantSystem,'noShear',true);  % Foam properties
+
+% Surfactant properties
+surf = getSurfactant(surfactantSystem); % Dissolution properties
+
+% Override partition coefficient from getSurfactant().
+surf.Cpart = 0.1;
+
+fluidF = addSimpleFoamProperties(fluid, ...
+    'foam', foam, ...
+    'cCrit', 2e-3, ...          % Critical concentration by weight
+    'cDecl', 2.5e3, ...         % Foam effect decline interval
+    'adsMax',surf.adsMax, ...   % Maximum adsorption 
+    'adsSat',surf.adsSat, ...   % Threshold concentration for adsorption
+    'surfingas', false, ...     % Gas-soluble surfactant
+    'mobMultIsLinear',true, ... % Is the mobility multiplier a linear function of gas velocity?
+    'usePermDep',false, ...     % Is foam strength dependent on absolute permeability?
+    'noShear', true);           % No velocity dependence
+
+model = GasWaterSurfactantFoamModel(G, rock, fluidF);
+model.OutputStateFunctions = [model.OutputStateFunctions,'PhaseFlux'];
+% No partitioning if commented
+model.fluid.Cpart = surf.Cpart;
+% Set up inital state
+state0 = initStateDeck(model, deck);
+
+%% Add foam-specific pproperties to initial state
+% This can be used e.g. to initialize a region around the injection well
+% with some surfactant dissolved in the water, as if a slug of surfactant
+% solution has already been injected.
+%
+% Initial surfactant:
+% X wt% in water
+% This should be equilibrated with surfactant adsorbed to rock in case of
+% adsorption, and with surfactant in gas in case of partitioning.
+%
+% Total concentration calculated based on concentration in water and
+% adsorbed. I.e., calculate the amount adsorbed when concentration in water
+% is X wt%, and add this to the total amount per m3 of space.
+%
+% Insert calculations below
+% Surfactant in water:
+%initialcW = . . .
+%mSf = model.operators.pv.*model.fluid.rhoWS.*model.fluid.bW(state0.pressure).* ...
+%    state0.s(:,1)*initialcW/(1-initialcW);
+%
+% Surfactant in gas:
+% initialcG = Cpart*initialcW;
+%
+% No initial gas saturation, so no additional surfactant in gas phase
+% Initial surfactant concentration, in kg per m3 pore volume
+
+% Adsorbed surfactant
+state0.cA = zeros(G.cells.num,1);
+%state0.cA = ones(G.cells.num,1)*model.fluid.surfads(initialcW);
+%mSf = mSf + state0.cA.*G.cells.volumes.*(1-model.rock.poro).*model.fluid.rhoRSft;
+
+% Initial volumetric concentration
+%c0 = (mSf)./model.operators.pv;
+%[state0.c, state0.cmax] = deal(ones(G.cells.num,1).*c0);
```

Comment:
For me this commented out line seems a bit confusing seeing as state.c is not a variable. Maybe delete?